gobblin-compaction/src/main/java/gobblin/compaction/event/CompactionSlaEventHelper.java

```diff
@@ -0,0 +1,89 @@
+package gobblin.compaction.event;
+
+import java.io.IOException;
+
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.mapreduce.Counter;
+import org.apache.hadoop.mapreduce.Counters;
+import org.apache.hadoop.mapreduce.Job;
+
+import gobblin.compaction.mapreduce.avro.AvroKeyDedupReducer;
+import gobblin.configuration.ConfigurationKeys;
+import gobblin.configuration.State;
+import gobblin.metrics.event.sla.SlaEventKeys;
+
+/**
+ * Helper class to populate sla event metadata in state.
+ */
+public class CompactionSlaEventHelper {
+
+  public static void populateState(State state , Job job, FileSystem fs) {
+    setDatasetUrn(state);
+    setPartition(state);
+    setDedupeStatus(state);
+    setPreviousPublishTime(state, fs);
+    setRecordCount(state, job);
+    setUpstreamTimeStamp(state, fs);
+  }
+
+  private static void setDatasetUrn(State state) {
+    state.setProp(SlaEventKeys.DATASET_URN_KEY, new Path(state.getProp(ConfigurationKeys.COMPACTION_DEST_DIR),
+        state.getProp(ConfigurationKeys.COMPACTION_TOPIC)).toString());
+  }
+
+  private static void setPartition(State state) {
+    state.setProp(SlaEventKeys.PARTITION_KEY, state.getProp(ConfigurationKeys.COMPACTION_JOB_DEST_PARTITION));
+  }
+
+  private static void setUpstreamTimeStamp(State state, FileSystem fs) {
+
+    String inputDirectory = state.getProp(ConfigurationKeys.COMPACTION_JOB_INPUT_DIR);
+    try {
+      FileStatus fileStatus = fs.getFileStatus(new Path(inputDirectory));
+      state.setProp(SlaEventKeys.UPSTREAM_TS_KEY, Long.toString(fileStatus.getModificationTime()));
+    } catch (IOException e) {
+    }
+  }
+
+  private static void setPreviousPublishTime(State state, FileSystem fs) {
+
+    Path compactionCompletePath =
+        new Path(state.getProp(ConfigurationKeys.COMPACTION_JOB_DEST_DIR),
+            ConfigurationKeys.COMPACTION_COMPLETE_FILE_NAME);
+
+    try {
+      FileStatus fileStatus = fs.getFileStatus(compactionCompletePath);
+      state.setProp(SlaEventKeys.PREVIOUS_PUBLISH_TS_KEY, Long.toString(fileStatus.getModificationTime()));
+    } catch (IOException e) {
+    }
+  }
+
+  private static void setDedupeStatus(State state) {
+    if (state.getPropAsBoolean(ConfigurationKeys.COMPACTION_DEDUPLICATE,
+        ConfigurationKeys.DEFAULT_COMPACTION_DEDUPLICATE)) {
+      state.setProp(SlaEventKeys.DEDUPE_STATUS_KEY, DedupeStatus.DEDUPED);
+
```

Comment:
What I meant is you probably want to return here, as line 70 overwrites the property you just set.